### ISSUE TITLE
Fix API definition mismatch

### ISSUE THREAD

**Issue by bafonins** [TheThingsNetwork/lorawan-stack#217]
**Summary:** 

The `api-definition.json` file is not synced with the current api definitions of the stack.
See https://github.com/TheThingsNetwork/lorawan-stack/blob/master/api/applicationserver.proto#L71 and https://github.com/TheThingsNetwork/lorawan-stack/blob/master/sdk/js/generated/api-definition.json#L188 as an example.

See tech dept issue: https://github.com/TheThingsNetwork/lorawan-stack/issues/32

**What do you see now?**

Deviation in the generated definitions from the actual api.

**What do you want to see instead?**

No deviation.

**How do you propose to implement this?**

Use `sdk/js/src/util/http-mapper.js` to update the api definition.

**Environment:**

Javascript SDK

**What can you do yourself and what do you need help with?**

Fix it

**Comment by htdvisser** [comment]
Generating the these files needs to become part of our build tooling. Let's discuss what exactly you need, because it doesn't look like you're using both `api.json` and the derived `api-definition.json`.

**Comment by bafonins** [comment]
> Generating these files needs to become part of our build tooling.

This is already specified here https://github.com/TheThingsNetwork/lorawan-stack/issues/30 in the `CI` section.

To clarify, currently, we parse the `generated/api.json` file to extract the data (http method, url, parameters, service name e.g. `ApplicationAccess`, service method names e.g.`CreateAPIKey`) that is needed to generate all stack api methods.

I would say that any altering of protos should also update the definitions that we have at the moment together with the markdown/swagger docs. Ff we decide to go with https://github.com/TheThingsNetwork/lorawan-stack/issues/32, then it should update the `es6` classes as well.

@kschiffer could you clarify how you updated the `api-definition.json` when renaming devices?

**Comment by kschiffer** [comment]
To update `api-definition.json`, run `sdk/js/dist/util/http-mapper.js`. This will ingest `api.json` and generate `api-definition.json` from it.

**Comment by bafonins** [comment]
@kschiffer what about the `generated/api.json` file?

**Comment by kschiffer** [comment]
This is what I thought gets generated from `make protos`, but apparently not. @htdvisser, can you clarify? I did not work on the generator for this.

**Comment by htdvisser** [comment]
As I wrote in https://github.com/TheThingsNetwork/lorawan-stack/issues/217#issuecomment-469205560, this still needs to become part of our build tooling. In order to implement that, I'd need to know what exactly you need and what not.

**Comment by bafonins** [comment]
@kschiffer how did you generate the file for initial commit for the js sdk?

**Comment by kschiffer** [comment]
It was provided to me by @htdvisser. It used to be generated with this command apparently
```
rm -rf tmp
mkdir tmp
docker run --rm -v $PWD:$PWD -w $PWD htdvisser/protocontainer -I $(dirname $PWD) -I $GOPATH/src --doc_out=$PWD/tmp --doc_opt=json,api.json --swagger_out=$PWD/tmp $PWD/api/*.proto
```
which does not seem to work anymore.

**Comment by johanstokking** [comment]
Closed by #231
